internal/service/organization/resource_organization_test.go

```diff
@@ -56,39 +63,71 @@ func testAccCheckMongoDBAtlasOrganizationExists(resourceName string) resource.Te
 			return fmt.Errorf("no ID is set")
 		}
 
+		if rs.Primary.Attributes["public_key"] == "" {
+			return fmt.Errorf("no public_key is set")
+		}
+
+		if rs.Primary.Attributes["private_key"] == "" {
+			return fmt.Errorf("no private_key is set")
+		}
+
 		ids := conversion.DecodeStateID(rs.Primary.ID)
 
-		organizationOptions := &matlas.OrganizationsListOptions{}
-		orgs, _, err := conn.Organizations.List(context.Background(), organizationOptions)
+		cfg := config.Config{
+			PublicKey:  rs.Primary.Attributes["public_key"],
+			PrivateKey: rs.Primary.Attributes["private_key"],
+			BaseURL:    acc.TestAccProviderSdkV2.Meta().(*config.MongoDBClient).Config.BaseURL,
+		}
+
+		ctx := context.Background()
+		clients, _ := cfg.NewClient(ctx)
+		conn := clients.(*config.MongoDBClient).AtlasV2
+
+		orgs, _, err := conn.OrganizationsApi.ListOrganizations(ctx).Execute()
 		if err == nil {
-			for _, val := range orgs.Results {
-				if val.ID == ids["org_id"] {
-					return fmt.Errorf("Organization (%s) still exists", ids["role_name"])
+			for _, val := range *orgs.Results {
+				if *val.Id == ids["org_id"] {
+					return nil
 				}
 			}
-			return nil
+			return fmt.Errorf("Organization (%s) doesn't exist", ids["org_id"])
 		}
 
 		return nil
 	}
 }
 
 func testAccCheckMongoDBAtlasOrganizationDestroy(s *terraform.State) error {
-	conn := acc.TestAccProviderSdkV2.Meta().(*config.MongoDBClient).Atlas
-
 	for _, rs := range s.RootModule().Resources {
 		if rs.Type != "mongodbatlas_organization" {
 			continue
 		}
 
+		if rs.Primary.Attributes["public_key"] == "" {
```

Comment:
as these 2 checks are done in a couple of places, don't know if it makes sense to extract them together

Reply:
updated